lib/actions/preload.js

```diff
@@ -178,11 +208,73 @@ Alternatively you can pass the --pin-device-to-release flag to pin only this dev
 		});
 };
 
+/**
+ * @param {import('balena-sdk').BalenaSDK} balenaSdk
+ * @param {string | number} appId
+ * @returns {Promise<import('balena-sdk').Application>}
+ */
+async function getAppWithReleases(balenaSdk, appId) {
+	return balenaSdk.models.application.get(appId, {
+		$expand: applicationExpandOptions,
+	});
+}
+
+async function prepareAndPreload(preloader, balenaSdk, options) {
+	const { ExpectedError } = require('../errors');
+
+	await preloader.prepare();
+	preloader.config = { deviceType: 'intel-nuc' };
+
+	const application = options.appId
+		? await getAppWithReleases(balenaSdk, options.appId)
+		: await selectApplication(preloader.config.deviceType);
+
+	/** @type {string} commit hash or the strings 'latest' or 'current' */
+	let commit;
+
+	// Use the commit given as --commit or show an interactive commit selection menu
+	if (options.commit) {
+		if (isCurrent(options.commit)) {
+			if (!application.commit) {
+				throw new Error(
+					`Unexpected empty commit hash for app ID "${application.id}"`,
+				);
+			}
+			// handle `--commit current` (and its `--commit latest` synonym)
+			commit = 'latest';
+		} else {
+			const release = _.find(application.owns__release, (r) =>
+				r.commit.startsWith(options.commit),
+			);
+			if (!release) {
+				throw new ExpectedError(
+					`There is no release matching commit "${options.commit}"`,
+				);
+			}
+			commit = release.commit;
+		}
+	} else {
+		// this could have the value 'current'
+		commit = await selectApplicationCommit(application.owns__release);
+	}
+
+	await preloader.setAppIdAndCommit(
+		application.id,
+		isCurrent(commit) ? application.commit : commit,
+	);
```

Comment:
`setAppIdAndCommit` is a new function added in balena-io-modules/balena-preload/pull/220